include/boost/geometry/util/normalize_spheroidal_coordinates.hpp

```diff
@@ -236,17 +237,22 @@ class normalize_spheroidal_coordinates
     }
 
 public:
-    static inline void apply(CoordinateType& longitude)
+    static inline void apply(CoordinateType& longitude, bool exact = true)
     {
         // normalize longitude
-        if (math::equals(math::abs(longitude), constants::half_period()))
+        CoordinateType const epsilon = CoordinateType(1e-07);
+        bool const is_integer = std::numeric_limits<CoordinateType>::is_integer;
+
+        if (exact || is_integer ? math::equals(math::abs(longitude), constants::half_period())
```

Comment:
`math::equals` is still not exact for floating point.
I'm not sure of the context, but the `1.e-7` looks a bit rough an arbitrary.

Reply:
Yes, it is indeed, I am not happy with this either. 

Selecting the right epsilon value depends on the threshold we want to set when declaring two points with longitude 180 to be equal. `1.e-7` is the smallest value so that the new tests are passing, actually set by [this test](https://github.com/boostorg/geometry/pull/1370/files#diff-6898dddc1f86c9a558e1a3871bf25eebfa1b492dcb7a0e7c7de1c86a78e400b5R200); if we only want to support [this test case](https://github.com/boostorg/geometry/pull/1370/files#diff-6898dddc1f86c9a558e1a3871bf25eebfa1b492dcb7a0e7c7de1c86a78e400b5R204)  a smaller value suffices. 
The latter case seems important since it could be the case the 180 degrees is converted to radian using the constants of the spheroid (that is the case of mysql).

Reply:
In any case, if we keep such an arbitrary value the selection should be documented in a comment.